api/server/v1/marshaler.go

```diff
@@ -494,3 +418,83 @@ func (x *DeleteResponse) MarshalJSON() ([]byte, error) {
 func (x *UpdateResponse) MarshalJSON() ([]byte, error) {
 	return json.Marshal(&dmlResponse{Metadata: CreateMDFromResponseMD(x.Metadata), Status: x.Status, ModifiedCount: x.ModifiedCount})
 }
+
+// MarshalJSON on read response avoid any encoding/decoding on x.Data. With this approach we are not doing any extra
+// marshaling/unmarshalling in returning the data from the database. The document returned from the database is stored
+// in x.Data and will return as-is.
+//
+// Note: This also means any changes in ReadResponse proto needs to make sure that we add that here and similarly
+// the openAPI specs needs to be specified Data as object instead of bytes.
+func (x *ReadResponse) MarshalJSON() ([]byte, error) {
+	resp := struct {
+		Data        json.RawMessage `json:"data,omitempty"`
+		Metadata    Metadata        `json:"metadata,omitempty"`
+		ResumeToken []byte          `json:"resume_token,omitempty"`
+	}{
+		Data:        x.Data,
+		Metadata:    CreateMDFromResponseMD(x.Metadata),
+		ResumeToken: x.ResumeToken,
+	}
+	return json.Marshal(resp)
+}
+
+func (x *SearchHit) MarshalJSON() ([]byte, error) {
+	resp := struct {
+		Data     json.RawMessage   `json:"data,omitempty"`
+		Metadata SearchHitMetadata `json:"metadata,omitempty"`
+	}{
+		Data:     x.Data,
+		Metadata: CreateMDFromSearchMD(x.Metadata),
+	}
+	return json.Marshal(resp)
+}
+
+type SearchHitMetadata struct {
+	CreatedAt *time.Time `json:"created_at,omitempty"`
+	UpdatedAt *time.Time `json:"updated_at,omitempty"`
+	DeletedAt *time.Time `json:"deleted_at,omitempty"`
```

Comment:
nit: search hit won't have `DeletedAt`

Reply:
let's remove it then to avoid any confusion.